yaml.go

```diff
@@ -0,0 +1,58 @@
+package runn
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Mainly respond to issue where the deep nested map becomes map[interface{}]interface{}
+// and issue where yaml.MapSlice is defined as nested.
+
+func yamlMarshal(in interface{}) ([]byte, error) {
+	return yaml.Marshal(in)
+}
+
+func yamlUnmarshal(in []byte, out interface{}) error {
+	return yaml.Unmarshal(in, out)
```

Comment:
@k1LoW 
Is there any reason why you didn't have it normalized within the `yamlUnmarshal` function?

Reply:
For the following reasons

- I didn't want to change the function signatures because I plan to revert to goccy/go-yaml
- On the other hand, the type of `out interface{}` was undefined, so it was difficult to implement without changing the signature

If the above can be resolved, it would be better to normalize within yamlUnmarshal as you say.

By the way, I created yamlUnmarshal to collect concrete implementations of YAML operations in yaml.go.
As a result, yamlUnmarshal and yamlUnmarshal are the same.

Reply:
And, the reason I created the `normalize()` is to bridge as much as possible the difference between goccy/go-yaml and go-yaml/yaml's Unmarshal function.

Reply:
> I didn't want to change the function signatures because I plan to revert to goccy/go-yaml

Although the direct cause has not been identified, some vars are not being deployed properly from the version that merged this PR.

Some parts of the process may not be normalized.

If this PR is going to revert with a trial content, should I just sit tight for once?

Reply:
> Although the direct cause has not been identified, some vars are not being deployed properly from the version that merged this PR.

That's not good.

Initially, this PR was created to resolve #103.

However, if it is not working well, we might revert this PR.

Reply:
Or give up the revert to goccy/go-yaml...

Reply:
Or we can wait for the following Issue to be resolved
https://github.com/goccy/go-yaml/issues/316

Reply:
@k2tzumi I have decided to revert this pr once.
#152